src/com/dabomstew/pkrandom/romhandlers/Gen3RomHandler.java

```diff
@@ -3413,4 +3413,46 @@ public BufferedImage getMascotImage() {
     public List<Integer> getAllConsumableHeldItems() {
         return Gen3Constants.consumableHeldItems;
     }
+
+    private static final Set<Type> physicalTypes = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            Type.NORMAL, Type.FIGHTING, Type.POISON, Type.GROUND, Type.FLYING, Type.BUG,
+            Type.ROCK, Type.GHOST, Type.STEEL)));
+
+    @Override
+    public List<Integer> getSensibleHeldItemsFor(TrainerPokemon tp, boolean consumableOnly, List<Move> moves, Map<Integer, List<MoveLearnt>> movesets) {
+        List<Integer> items = new ArrayList<>();
+        items.addAll(Gen3Constants.generalPurposeConsumableItems);
+        if (!consumableOnly) {
+            items.addAll(Gen3Constants.generalPurposeItems);
+        }
+        int[] pokeMoves = RomFunctions.getMovesAtLevel(tp.pokemon.number, movesets, tp.level);
+        for (int moveIdx : pokeMoves) {
+            Move move = moves.get(moveIdx);
+            if (move == null) {
+                continue;
+            }
+            if (physicalTypes.contains(move.type) && move.power > 0) {
+                items.add(Gen3Constants.liechiBerry);
+                if (!consumableOnly) {
+                    items.addAll(Gen3Constants.typeBoostingItems.get(move.type));
+                    items.add(Gen3Constants.choiceBand);
+                }
+            }
+            if (!physicalTypes.contains(move.type) && move.power > 0) {
+                items.add(Gen3Constants.petayaBerry);
+                if (!consumableOnly) {
+                    items.addAll(Gen3Constants.typeBoostingItems.get(move.type));
+                }
+            }
+        }
+        if (!consumableOnly) {
+            List<Integer> speciesItems = Gen3Constants.speciesBoostingItems.get(tp.pokemon.number);
+            if (speciesItems != null) {
+                for (int i = 0 ; i < 8; i ++) {  // Increase the likelihood of using species specific items.

```

Comment:
I think it's reasonable to boost the probability for certain items to appear, but is 8 the appropriate multiplier for this? Not just for the species specific ones, but also the weakness reducing berries in Gen 4/5. I would put it maybe at 4, which would still make them significantly more likely but not overwhelmingly so. Feel free to give your own thoughts, though.

Reply:
The item pool is pretty large. Even with the boost of 8x, the supereffective berries only show up about a dozen times each across the 1304 pokemon in Pokemon Black. In soulsilver, the numbers creep up into the high teens or so (out of 1776 pokemon). The chilan berry (unboosted and only for neutral effective pokemon) shows up 20-30 times (as a comparison). Bright powder (an arbitrary "general purpose" item) also shows up 20-30 times.

For species specific items, with the 8x boost, they typically show up 0 or 1 times when giving random items to all trainer pokemon in the game.

In short, with the 8x multiplier, the weakness reducing berries (8x only applies to double weakness) show up less often than a general purpose item. With the 8x multiplier, species specific items are now likely to happen, but only once or twice across an entire run. I played with a few multipliers until this one "felt" right.

Reply:
Rather than looking at how many times the boosted-probability items appear in total, I think the more interesting metric to look at is: for Pokemon that are affected by this, how high is the probability that they will get the boosted-probability item? A player probably won't keep track of how many total Yache Berries they've run into in a single run, but if they encounter two Dragonites in the same run that both have Yache Berries they will probably notice. It shouldn't be too predictable.

I tried running a few tests on Emerald, Soul Silver and Black, to see how large the item pools/probabilities were for Pokemon where boosted-probability items were picked:

Emerald: ~35-50 (~16-23%)
Soul Silver: ~60-80 (~10-13%)
Soul Silver (consumable only): ~35-50 (~16-23%)
Black: ~65-95 (~8-12%)
Black (consumable only): ~40-55 (~15-20%)

This is not accounting for the Pokemon that have two quad weaknesses, where the percentages would be doubled (although their pools would be on the larger end so we wouldn't be looking at 46% for SS/consumable only - but 32% is still pretty high). [Here are the raw numbers from a single run of each, if you're interested.](https://pastebin.com/jLXhk7hr)

I think some of these are fine while others may warrant adjusting. 20% is too high in my opinion. Tweaking these on a per-generation basis as well as depending on the "Consumable Only" setting seems reasonable to me. I would suggest something like:

Gen 3: 6
Gen 4 (consumable only): 6
Gen 5 (consumable only): 6

What do you think?

Reply:
That is excellent analysis. I will implement your suggestions. I agree that 20ish percent is a bit high for any given item to appear, and 10ish percent is a bit of a nicer middle ground - common, but not too common.